From afe082f97979aeca2ee9157ed1349fe300b31255 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Fri, 2 Nov 2012 15:31:23 +0800
Subject: [PATCH 093/281] ENGR00232210 Wifi: support two interface concurrent

1. let wpa_supplicant control two interfaces by running one process
for each interface separately.
2. add dhcpcd svc for p2p0 interface.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 imx53_smd/init.freescale.rc | 4 ++--
 imx5x/init.rc               | 5 +++++
 imx6/etc/init.rc            | 5 +++++
 imx6/sabresd/init.rc        | 4 ++--
 4 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/imx53_smd/init.freescale.rc b/imx53_smd/init.freescale.rc
index 816c0eb..206f856 100755
--- a/imx53_smd/init.freescale.rc
+++ b/imx53_smd/init.freescale.rc
@@ -70,8 +70,8 @@ service hciattach /system/bin/logwrapper /system/bin/hciattach -n -s 115200 /dev
     oneshot
 
 service p2p_supplicant /system/bin/wpa_supplicant \
-    -Dnl80211 -puse_p2p_group_interface=0  -e/data/misc/wifi/entropy.bin \
-    -iwlan0 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
     class late_start
     disabled
     oneshot
diff --git a/imx5x/init.rc b/imx5x/init.rc
index 42f21e4..048c9d0 100644
--- a/imx5x/init.rc
+++ b/imx5x/init.rc
@@ -539,6 +539,11 @@ service dhcpcd_wlan0 /system/bin/dhcpcd -ABKL
     disabled
     oneshot
 
+service dhcpcd_p2p /system/bin/dhcpcd -ABKL
+    class late_start
+    disabled
+    oneshot
+
 service dhcpcd_eth0 /system/bin/dhcpcd -AL
     class late_start
     disabled
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 39c62e3..85acea9 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -559,6 +559,11 @@ service dhcpcd_wlan0 /system/bin/dhcpcd -ABKL
     disabled
     oneshot
 
+service dhcpcd_p2p /system/bin/dhcpcd -ABKL
+    class late_start
+    disabled
+    oneshot
+
 service dhcpcd_eth0 /system/bin/dhcpcd -ABKL
     class late_start
     disabled
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index 1d4dcbc..e9b5581 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -95,8 +95,8 @@ on boot
     setprop wlan.interface wlan0
 
 service p2p_supplicant /system/bin/wpa_supplicant \
-    -Dnl80211 -puse_p2p_group_interface=0  -e/data/misc/wifi/entropy.bin \
-    -iwlan0 -c/data/misc/wifi/wpa_supplicant.conf
+    -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
+    -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
     class late_start
     disabled
     oneshot
-- 
1.8.0

